### PR TITLE
Moved commerce hostname to ENTRYPOINT 

### PR DESCRIPTION
**Pull request by Barsonax**
Moved commerce hostname to ENTRYPOINT so that it can be defined in the compose file. This makes it easier to change the hostname between environments and keeps it out of the build image.

### CODE REVIEW

File: xc/commerce/Dockerfile [hunk 1]
```diff
@@ -215,34 +161,17 @@ RUN $CommerceServicesPathCollection = @('C:\\inetpub\\wwwroot\\CommerceAuthoring
...
     } 
 
 # Configure the business tools
 RUN $pathToJson = $(Join-Path -Path 'C:\\inetpub\\wwwroot\\SitecoreBizFx' -ChildPath "assets\\config.json"); `
     $originalJson = Get-Content $pathToJson -Raw | ConvertFrom-Json; `
-    $originalJson.EnvironmentName = $Env:ENVIRONMENT_NAME; `
-    $originalJson.EngineUri = ('https://{0}:5000' -f $Env:HOST_NAME); `
-    $originalJson.IdentityServerUri = ('https://{0}:5050' -f $Env:HOST_NAME); `
-    $originalJson.BizFxUri = ('https://{0}:4200' -f $Env:HOST_NAME); `
+    $originalJson.EnvironmentName = $Env:ENVIRONMENT_NAME; ``
```
Comment (joostmeijles) [inline xc/commerce/Dockerfile]: minus `

File: xc/commerce/UpdateHostname.ps1 [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+param(
+    [Parameter(Mandatory=$true)]
+    [String]$commerceHostname
+)
+
+$ErrorActionPreference = 'Stop'
+$ProgressPreference = 'SilentlyContinue'
+
+Function UpdateBindingCertificate() {
```
Comment (joostmeijles) [inline xc/commerce/UpdateHostname.ps1]: function

File: xc/commerce/UpdateHostname.ps1 [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            $p.AllowedCorsOrigins = $plumberAllowedOrigins
+        }
+    }
+
+    $json = ConvertTo-Json $json -Depth 100
+    Set-Content $configPath -Value $json -Encoding UTF8
+    Write-Host "Done patching $configPath!" -ForegroundColor Green
+}
+
+Function UpdateCommerceConfig() {
```
Comment (joostmeijles) [inline xc/commerce/UpdateHostname.ps1]: Function of function?

File: xc/commerce/UpdateHostname.ps1 [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+Write-Host "Succesfully created the certificate!. Updating bindings..." -ForegroundColor Green
+
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceOps_Sc9' -port 5015
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceShops_Sc9' -port 5005
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceAuthoring_Sc9' -port 5000 
+UpdateBindingCertificate -certificate $certificate -bindingName 'CommerceMinions_Sc9'-port 5010
+UpdateBindingCertificate -certificate $certificate -bindingName 'SitecoreBizFx' -port 4200
+UpdateBindingCertificate -certificate $certificate -bindingName 'SitecoreIdentityServer' -port 5050
+
+#Modify the config with the new hostnames
```
Comment (joostmeijles) [inline xc/commerce/UpdateHostname.ps1]: Maybe a little more comment describing what happens here?

File: xc/commerce/UpdateHostname.ps1 [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+[string]$identityServerUrl = $baseUrl + ":5050"
+[string]$engineServerUrl = $baseUrl + ":5000"
+[string]$bizFxUrl = $baseUrl + ":4200"
+[string]$plumberUrl = $baseUrl + ":4000"
+$redirectUrls = [string[]]@(
+    ($bizFxUrl),
+    ($bizFxUrl + "/?")
+)
+$allowedOrigins = [string[]]@(
+    "http://$sitecoreHostname",
```
Comment (joostmeijles) [inline xc/commerce/UpdateHostname.ps1]: I think this should remain to be set to the default `http://sitecore` iso `http://sitecore.local` as we do not change the Sitecore domain name.

Comment (Barsonax) [reply xc/commerce/UpdateHostname.ps1]: Actually sitecore can now be accessed at sitecore.local in our images.

https://github.com/avivasolutionsnl/sitecore-docker/pull/99/files#diff-d591119415fa54e2ca1f21deca0591eaR83

Comment (joostmeijles) [reply xc/commerce/UpdateHostname.ps1]: No need to update the Sitecore sites config for that?

**Pull request merged**